### ISSUE TITLE
When linter loop limit is hit, should report an error/exception

### ISSUE THREAD

**Issue by barrywhart** [sqlfluff/sqlfluff#1386]
When the loop limit is hit, `sqlfluff fix` logs a warning:
```
Loop limit on fixes reached
```

This is a pretty "quiet" notification that can mask issues. Suggest we change this:
* Increase the default loop limit from 10 to 25 iterations.
* Log an ERROR and/or raise an exception. If an exception, suggest (maybe?) raising one that is caught and included in the list of lint issues, not one that stops the `fix` run. Open to other ideas, but I think this is the right tradeoff between raising visibility of what is _likely_ an infinite loop in the linter run while still giving the user useful `fix` output.

Related to #1344, among others.
